iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -197,6 +197,47 @@ private async void ConnectionStatusChangeHandler(ConnectionStatus status, Connec
             }
         }
 
+        private async Task SubscribeToTwinUpdateNotificationsAsync(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (IsDeviceConnected)
+                {
+                    await s_deviceClient.SetDesiredPropertyUpdateCallbackAsync(HandleTwinUpdateNotificationsAsync, cancellationToken);

```

Comment:
Why does this operation need to be in a while loop? Wouldn't one check of the cancellation token and one attempt at setting the desired property callback be enough?

Reply:
Ideally, yes, but we don't want to set the callback unless the client is connected.

Reply:
Wouldn't that have been solved by keeping the explicit call to OpenAsync?

Reply:
Not really, disconnections can happen at any point, right